### PR TITLE
Release 0.3.0

### LINKED ISSUE #18: Simplifying amap usage
Reading through the docs it came to my mind that the `async with` usage of `amap` doesn’t “look necessary” from a UX perspective.

When using it we always have to do *two things*:

- Enter a context with `async with` (*looks* like an implementation detail)
- Iterate over results with `async for`

As a user really only care about the second operation.

So what if we moved `async with` inside the implementation of amap, so that users can just do...

```python
results = [result async for result in amap(process, items)]
```

### PR DESCRIPTION
**Pull request by florimondmanca**
### Changed

- Update `anyio` dependency to v3 (previously v1). (Pull #25)
  - _NB: no API change, but dependency mismatches may occur. Be sure to port your codebase to anyio v3 before upgrading `aiometer`._

### Added

- Add support for Python 3.6 (installs the `contextlib2` backport library there). (Pull #26)
- Officialize support for Python 3.9. (Pull #26)

### CODE REVIEW

**Review by graingert — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -70,8 +68,10 @@ _Try this code interactively using [IPython](https://ipython.org/install.html)._
 
 ## Installation
 
+_This project is still maturing. Be sure to pin any dependencies to the latest minor._
```
Comment (graingert) [inline README.md:71]: might want to update the trove classifiers to reflect this

Comment (florimondmanca) [reply README.md:71]: We are already pointing at `Status :: Alpha`. Anything else we should do?

Comment (graingert) [reply README.md:71]: `"Development Status :: 4 - Beta"`

Comment (florimondmanca) [reply README.md:71]: True. :)

**Review by graingert — APPROVED** [review]

**Pull request merged**